In [9]:
import pandas as pd

from sklearn.model_selection import train_test_split
from sklearn.pipeline import make_pipeline
from sklearn.feature_extraction.text import CountVectorizer
from sklearn.linear_model import LogisticRegression

from hyper.nlp_model import NLPModel

Read in the data and then split into true and false so we have equal numbers of trues and falses in the train and test.

In [6]:
data = pd.read_json("../data/clean.json")
trues = data[data["label"]]
falses = data[data["label"] == False]

Split the trues and falses into test and train

In [7]:
t_train, t_test = train_test_split(trues, test_size=0.5, random_state=13)
f_train, f_test = train_test_split(falses, test_size=0.5, random_state=13)
train = pd.concat([t_train, f_train])
test = pd.concat([t_test, f_test])

In [8]:
reviews_train = train["content"]
reviews_test = test["content"]
y_train = train["label"]
y_test = test["label"]

model = NLPModel()
model.fit_vectorizer(reviews_train)

X_train = model.transform_vectorizer(reviews_train)
X_test = model.transform_vectorizer(reviews_test)
model.train(X_train, y_train)
print(model.report_accuracy(X_test, y_test))

0.7213622291021672


This is an accuracy of 0.7213622291021672 which is suspiciously high. It gets this every time.

In [14]:
pipe = make_pipeline(
    CountVectorizer(),
    LogisticRegression(max_iter=300),
)
pipe.fit(reviews_train, y_train)
pipe.score(reviews_test, y_test)

0.7213622291021672

This is the same as above, so all of the stuff in my `NLPModel` is doing is this.